packages/web-app-files/tests/unit/helpers/collaborators.spec.js

```diff
@@ -0,0 +1,261 @@
+        allowSharePerm: true,
+        expectedRole: 'viewer',
+        expectedPermissions: ['read', 'share']
+      },
+      {
+        bitmask: 3,
+        isFolder: false,
+        allowSharePerm: false,
+        expectedRole: 'editor',
+        expectedPermissions: ['read', 'update']
+      },
+      {
+        bitmask: 15,
+        isFolder: true,
+        allowSharePerm: false,
+        expectedRole: 'editor',
+        expectedPermissions: ['read', 'update', 'create', 'delete']
+      },
+      {
+        bitmask: 31,
+        isFolder: true,
+        allowSharePerm: true,
+        expectedRole: 'editor',
+        expectedPermissions: ['read', 'update', 'create', 'delete', 'share']
+      }
+    ])(
+      'should return the first matching role if the given bitmask equals the role permissions bitmask',
+      input => {
+        // assume a case when share permission is allowed and resource is not folder and provided bitmask is 17
+        // two roles with following permissions are available from the roleDefinition helper
+        // - viewer with permissions ['read', 'share']
+        // - editor with permissions ['read', 'update', 'share']
+        // with the first role, the sum of permission bitmask (1 + 16) matches provided bitmask (17)
+        // so the function has found a match and returns the role {viewer}
+        expect(bitmaskToRole(input.bitmask, input.isFolder, input.allowSharePerm)).toMatchObject({
+          name: input.expectedRole,
+          permissions: input.expectedPermissions
+        })
+      }
+    )
+    it.each(['17', 17.2, '17.8', ['17.8']])('should parse provided bitmask to integer', bitmask => {
+      expect(bitmaskToRole(bitmask, false, true)).toMatchObject({
+        name: 'viewer'
+      })
+    })
+    it.each([true, 0, {}])('should return advanced role for invalid bitmask values', bitmask => {
+      expect(bitmaskToRole(bitmask, false, true)).toMatchObject({
+        name: 'advancedRole'
+      })
+    })
+  })
+
+  describe('check permission', () => {
+    it.each([
+      permissionsBitmask.read + permissionsBitmask.update,
+      (permissionsBitmask.read + permissionsBitmask.update).toString(),
+      permissionsBitmask.update,
+      permissionsBitmask.update.toString()
+    ])('should return true if given bitmask is found in permissions bitmask', bitmask => {
+      expect(checkPermission(bitmask, 'update')).toBeTruthy()
+    })
+    it('should return false if given bitmask does not equal permissionBitmask', () => {
+      expect(checkPermission(permissionsBitmask.read, 'update')).toBeFalsy()
+    })
+    it('should return false if given permission is not found in permissions bitmask', () => {
+      expect(checkPermission(1, 'modify')).toBeFalsy()
+    })
+    it.each([
+      // seems an edge case to me, should this be fixed?
+      // if permission is valid and bitmask is `true` function returns truthy value
+      { bitmask: true, permission: 'read', expected: true },
```

Comment:
@kulmann what do you think?

Reply:
Hmm. we should do two things:
1. convert the helper to typescript and allow string and number as types for the bitmask
2. as we won't have strict type checks in the near future it's probably still a good idea to add a typecheck into the code that only string and number are allowed. i.e. return `false` when the bitmask is neither a string nor a number.

If you want to you could add the type check (the second bullet) to the helper and adapt the test here accordingly. Otherwise please open a ticket and we can do it then. Thanks for the pointer! 👍

Reply:
i've added a type check for the functions, please reveiw again